Source/WebGPU/WGSL/CMakeLists.txt

```diff
@@ -0,0 +1,214 @@
+# Copyright 2024 Google LLC
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+# Build wgslc.
+# For simplicity, build it as a monolithic executable.
+
+# Target 'wgsl-types' generates TypeDeclarations.h in the build tree.
+add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/TypeDeclarations.h
+    COMMAND /usr/bin/env ruby
+        ${CMAKE_CURRENT_SOURCE_DIR}/generator/main.rb
+        ${CMAKE_CURRENT_SOURCE_DIR}/TypeDeclarations.rb
+        ${CMAKE_CURRENT_BINARY_DIR}/TypeDeclarations.h
+    DEPENDS
+        ${CMAKE_CURRENT_SOURCE_DIR}/generator/main.rb
+        ${CMAKE_CURRENT_SOURCE_DIR}/TypeDeclarations.rb
+    )
+add_custom_target(wgsl-types
+    ALL
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/TypeDeclarations.h)
+
+
+# Configure exectuable 'wgslc'
+
```

Comment:
I think the comment is redundant, as it is already clear what the intention is. Also IIRC one empty line would follow the WebKit CMake style better:

```suggestion

```